src/openvic-simulation/country/CountryInstance.cpp

```diff
@@ -2,59 +2,158 @@
+#include "openvic-simulation/map/MapInstance.hpp"
+#include "openvic-simulation/politics/Ideology.hpp"
+#include "openvic-simulation/research/Invention.hpp"
+#include "openvic-simulation/research/Technology.hpp"
 
 using namespace OpenVic;
 
+static constexpr colour_t ERROR_COLOUR = colour_t::from_integer(0xFF0000);
+
 CountryInstance::CountryInstance(
 	CountryDefinition const* new_country_definition,
 	decltype(technologies)::keys_t const& technology_keys,
 	decltype(inventions)::keys_t const& invention_keys,
-	decltype(upper_house)::keys_t const& ideology_keys
-) : country_definition { new_country_definition },
-	primary_culture { nullptr },
-	religion { nullptr },
-	ruling_party { nullptr },
-	last_election {},
+	decltype(upper_house)::keys_t const& ideology_keys,
+	decltype(pop_type_distribution)::keys_t const& pop_type_keys
+) : /* Main attributes */
+	country_definition { new_country_definition },
+	colour { ERROR_COLOUR },
 	capital { nullptr },
+	country_flags {},
+	civilised { false },
+	releasable_vassal { true },
+	owned_provinces {},
+	controlled_provinces {},
+	core_provinces {},
+	states {},
+
+	/* Production */
+
+	/* Budget */
+	cash_stockpile { 0 },
+
+	/* Technology */
+	technologies { &technology_keys },
+	inventions { &invention_keys },
+	current_research { nullptr },
+	invested_research_points { 0 },
+	expected_completion_date {},
+	research_point_stockpile { 0 },
+	daily_research_points { 0 },
+	national_literacy { 0 },
+	tech_school { nullptr },
+
+	/* Politics */
+	national_value { nullptr },
 	government_type { nullptr },
+	last_election {},
+	ruling_party { nullptr },
+	upper_house { &ideology_keys },
+	reforms {},
+	suppression_points { 0 },
+	infamy { 0 },
 	plurality { 0 },
-	national_value { nullptr },
-	civilised { false },
+	revanchism { 0 },
+
+	/* Population */
+	primary_culture { nullptr },
+	accepted_cultures {},
+	religion { nullptr },
+	total_population { 0 },
+	national_consciousness { 0 },
+	national_militancy { 0 },
+	pop_type_distribution { &pop_type_keys },
+	national_focus_capacity { 0 },
+
+	/* Trade */
+
+	/* Diplomacy */
+	total_rank { 0 },
 	prestige { 0 },
-	upper_house { &ideology_keys },
-	technologies { &technology_keys },
-	inventions { &invention_keys } {}
+	prestige_rank { 0 },
+	industrial_power { 0 },
+	industrial_rank { 0 },
+	military_power { 0 },
+	military_rank { 0 },
+	diplomatic_points { 0 },
+
+	/* Military */
+	generals {},
+	admirals {},
+	armies {},
+	navies {},
+	regiment_count { 0 },
+	mobilisation_regiment_potential { 0 },
+	ship_count { 0 },
+	total_consumed_ship_supply { 0 },
+	max_ship_supply { 0 },
+	leadership_points { 0 },
+	war_exhaustion { 0 } {}
 
 std::string_view CountryInstance::get_identifier() const {
-	return country_definition != nullptr ? country_definition->get_identifier() : "NULL";
+	return country_definition->get_identifier();
 }
 
-bool CountryInstance::add_accepted_culture(Culture const* new_accepted_culture) {
```

Comment:
Where did this method go?
I can't find it in the .cpp anymore.

Reply:
It's generated by the `ADD_AND_REMOVE` macro in `CountryInstance.cpp` which makes the add and remove functions for owned provinces, controlled provinces, core provinces, states and accepted cultures.